### PR TITLE
[LM-9399] All Post Type UI

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**
- Added Image, Video, Carousel, Document Post UI

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    showPin,
+    threeDotIcon,
+    pinIcon,
+    onThreeDotClick
+  }: PostUI
+) => {
+  const [showModal, setShowModal] = useState(false);
+  const [modalPosition, setModalPosition] = useState({x:0, y:0});
+  const [pinPost, setPinPost] = useState(showPin)
+
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:36]: Add comment to explain function

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    threeDotIcon,
+    pinIcon,
+    onThreeDotClick
+  }: PostUI
+) => {
+  const [showModal, setShowModal] = useState(false);
+  const [modalPosition, setModalPosition] = useState({x:0, y:0});
+  const [pinPost, setPinPost] = useState(showPin)
+  
+  const defaultThreeDotIcon = () => {
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:37]: Add comment to explain function

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        return (
+        <Image source={require('../../../assets/images/three_dots3x.png')} resizeMode={'contain'}  style={styles.iconSize}/>
+        )
+  }
+  const defaultPinIcon = () => {
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:42]: Add a comment to explain the function

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        return (
+        <Image source={require('../../../assets/images/pin_icon3x.png')} resizeMode={'contain'}  style={styles.iconSize}/>
+        )
+  }
+  const onThreedotsClick = (event: any) => {
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:47]: Add a comment to explain the function

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      const {pageX, pageY} = event.nativeEvent;
+      setShowModal(true)
+      setModalPosition({x:pageX, y:pageY});
+      onThreeDotClick && onThreeDotClick()
+  }
+
+  const closePostActionListModal =() => {
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:54]: Add a comment to explain the function

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    setShowModal(false)
+  }
+
+  const handlePinPost = () => {
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:58]: Add a comment to explain the function

File: FeedSX/components/LMPost/PostHeader/index.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                </View>
+            </View>
+        </View>
+
+        {/* Top right action buttons */}
+        <View style={[styles.topRightView, pinPost && {justifyContent: 'space-between'}]}>
+         {pinPost && <LMIcon displayIcon={pinIcon? pinIcon : defaultPinIcon} />}
+          <LMIcon displayIcon={threeDotIcon? threeDotIcon : defaultThreeDotIcon} onIconPress={onThreedotsClick} />
+        </View>
+       <PostActionListModal
```
Comment (jaixchaudhary) [inline FeedSX/components/LMPost/PostHeader/index.tsx:95]: Add a comment to explain the component

File: package.json [hunk 8]
```diff
@@ -10,6 +10,7 @@
     "test": "jest"
   },
   "dependencies": {
+    "@flyerhq/react-native-link-preview": "^1.6.0",
```
Comment (jaixchaudhary) [inline package.json:13]: remove caret(^) from all the dependencies

**Review by jaixchaudhary — APPROVED** [review]

File: FeedSX/components/LMPost/index.tsx [hunk 9]
```diff
@@ -49,12 +84,47 @@ const LMPost = ({
...
+
+      {/* Post media section */}
+      {attachment_type && attachment_type?.length > 1 ?
+      attachment_type.filter((curr) => curr === 1 || curr ===2).length >= 2 ?
+      <CarouselPost carauselActiveItemColor={carauselActiveItemColor} carauselInActiveItemColor={carauselInActiveItemColor} carouselPaginationStyle={carouselPaginationStyle} />
+      : 
+      <>{attachment_type?.find((e) => e===1) && <ImagePost mediaUrl={mediaUrl} />}
+      {attachment_type?.find((e) => e===2) && <VideoPost />}
+      {attachment_type?.find((e) => e===3) && <DocumentPost />}</>
+      :
```
Comment (ishaan1607) [inline FeedSX/components/LMPost/index.tsx:123]: what is `e`?

File: FeedSX/components/LMPost/index.tsx [hunk 10]
```diff
@@ -49,12 +84,47 @@ const LMPost = ({
...
+      <>{attachment_type && attachment_type?.[0] === 1 && <ImagePost mediaUrl={mediaUrl} />}
+      {attachment_type && attachment_type?.[0] === 2 && <VideoPost />}
+      {attachment_type && attachment_type?.[0] === 3 && <DocumentPost />}
+      {attachment_type && attachment_type?.[0] === 4 && <LinkPost />}</>}
```
Comment (ishaan1607) [inline FeedSX/components/LMPost/index.tsx:127]: use enums instead of 1, 2, 3 or 4

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 11]
```diff
@@ -17,11 +17,22 @@ const UniversalFeed = () => {
...
+        footerTextStyle: {},
+        authorName: 'Theresa Webb',
+        postedTime: '2h',
+        showEdited: true,
+        showLabel: true,
+        labelType: 'Admin',
+        showPin: false,
+        mediaUrl:'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSJhNhTcdpoN6c-rzLj336_o2WpgLgeqirPchSSBerB&s',
+        postText:`Here is a list of social media tools to help you get started with your marketing initiatives.`,
+        attachment_type: [3]
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:29]: why hard-coding this values?

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: FeedSX/customModals/PostActionListModal/index.tsx [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+}
+type Props = PostActionListProps & PostUI
+
+const PostActionListModal: React.FC<Props> = props => {
+  const{position, visible, closeModal, handlePin} = props as PostActionListProps;
+  const{modalStyle, modalTextStyle, modalBackdropColor} = props as PostUI;
+  const menuItems = [
+    {id : 1, Title: 'Delete'},
+    {id : 2, Title: 'Pin'},
+    {id : 5, Title: 'Edit'}]
```
[the remark this answers is not in the document]
Comment (prabha-likeminds) [reply FeedSX/customModals/PostActionListModal/index.tsx:22]: It is dummy array, will be removed once api will be implemented.

File: FeedSX/screens/UniversalFeed/index.tsx (hunk 11 above)
Comment (prabha-likeminds) [reply FeedSX/screens/UniversalFeed/index.tsx:29]: These are also dummy params, will be removed after api implementation.

File: FeedSX/Models/PostModel/index.tsx [hunk 13]
```diff
@@ -34,8 +34,8 @@ export interface PostStateProps {
     // post media
     mediaUrl?: string,
     postText?: string,
-    postTextStyle?: TextStyle
-    attachment_type?: Array<number>
+    postTextStyle?: TextStyle,
+    postAttachments?: Array<any>,
```
Comment (ishaan1607) [inline FeedSX/Models/PostModel/index.tsx:38]: why `any`?

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 14]
```diff
@@ -1,39 +1,104 @@
...
+    // const UUID = await AsyncStorage.getItem('userUniqueID');
+    const UUID = '10003';
+
+    let payload = {
+      userUniqueId: UUID, // user unique ID
+      userName: 'abc', // user name
+      isGuest: false,
     };
+    // calling initiateUser API
+    let res = await dispatch(initAPI(payload) as any);
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:31]: rename to `initiateResponse`

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 15]
```diff
@@ -1,39 +1,104 @@
...
+
+  // this functions gets universal feed data
+  async function fetchFeed() {
+    let payload = {
+      page: 1,
+      pageSize: 10,
+    };
+    // calling getFeed API
+    let res = await dispatch(
+      getAllFeed(
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:49]: rename to `getAllFeed()` to `getFeed()`. All these functions name should be same as Data Layer.

File: FeedSX/store/actions/feed.ts [hunk 16]
```diff
@@ -1,49 +1,69 @@
...
-//       });
-//     } catch (error) {
-//       Alert.alert(`${error}`)
-//     }
-//   };
+// initiateUser API action
+export const initAPI = (payload?: any) => async (dispatch: Dispatch) => {
+  const { userUniqueId,userName, isGuest} = payload
+  try {
+    const params = InitiateUserRequest.builder().setUUID(userUniqueId).setIsGuest(isGuest).setUserName(userName).build()
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:22]: rename `params`  to `request`

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**